lib/Doctrine/ORM/Query/ParameterTypeInferer.php

```diff
@@ -53,7 +53,7 @@ public static function inferType($value)
             return Type::BOOLEAN;
         }
 
-        if ($value instanceof \DateTime) {
+        if ($value instanceof \DateTime || $value instanceof \DateTimeImmutable) {
```

Comment:
I know but `\DateTimeImmutable` too. So when I do not pass instance of `\DateTime` in PHP 5.4 or older then it will fail on the second part of this condition, right?